test/compile/selection/scales.test.ts

```diff
@@ -36,11 +37,11 @@ describe('Selection + Scales', function() {
             },
             color: {
               field: "symbol", type: "nominal",
-              scale: {domain: {selection: "brush2"}}
+              scale: {domain: {selection: "brush2"} as Domain}
```

Comment:
@arvind    Currently 
```
export type SelectionDomain = {selection: string, field: string} | {selection: string, encoding: string};
```

so `{selection: "brush2"}` actually does not satisfy any case.  Not sure if the test spec here is intended.  (Same for the `{selection: "brush3"}` below.)